Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -93,8 +97,46 @@ public final CustomUIPane getPane() {
      * change.
      *
      * This method should only be called after the Block's constructor is done.
+     * 
+     * This method will invalidate the Block even if the state did not change.
+     */
+    public void invalidateConnectionState() {
+    }
+
+    /**
+     * Does the same as invalidateConnectionState(), but cascading down to
+     * other blocks which are possibly also (indirectly) affected by the state
+     * change.
+     * 
+     * @param state
+     *            The newest visual state
+     */
+    public void invalidateConnectionStateCascading(int state) {
+        if (!connectionStateIsUpToDate(state)) {
```

Comment:
Hier wordt zo te zien bovendien de aanname gedaan dat als dit blok up to date is, ook alle blokken downstream up to date zijn.